### DISCUSSION TITLE
Add the possibility of changing attributes

### DISCUSSION THREAD

**Discussion by TkachenkoViacheslav** [drmohundro/SWXMLHash#260] [Ideas]
It will be greate to have the possibilty of changing/adding xml element attributes.
And if you'll declare **init** method of the `public struct XMLAttribute` as **public** - that's will be enough!

Real example: in my case I have to edit html string and change some attributes of the specific xml elements in it:

```
let xml = XMLHash.parse(str)
if let element = xml.element {
    element.children.forEach { content in
        guard let element = content as? XMLElement else { return }
        element.allAttributes["someAttribute"] = XMLAttribute(name: "someAttribute", text: "newValue")
    }
    let newStr = xml.description
}
```

**Comment by drmohundro** [comment]
Hi @Slavik36, thanks for dropping a note! I'm not totally opposed to making some of the objects public, but I'd want to think about what it might look like to expose the other "XML building" methods... for example, `addElement` off of `XMLElement` and so on. And then, changing `XMLHash` such that it could receive an `XMLElement` that would serve as the data that could then be indexed. Does that make sense?

**Comment by TkachenkoViacheslav** [comment]
@drmohundro, thanks for you quick response!
Here're my thoughts on this:
- XMLAttribute is an inner part of the XMLElement and it has no relationships to the indexes (XMLIndexer).
- And even in the future we will no need in adding attribute to the parsed document independently from the XMLElement.

Thus, we can split mutating of the xml into the two parts (features): working with the attributes of the XMLElement and working with the whole XML document.
So, speaking about attributes, we can forgot about work with the whole XML document and the only one case in which I see the need of using indexes of the attributes is a feature "to store initial order of attributes as in the source string" (to prevent: `<elem a1="1" a2="2"/>` --> `<elem a2="2" a1="1"/>`) - and it will be greate, but not as important as modifying value of the attributes.

**Reply by drmohundro** [reply]
Well, it wouldn't be too difficult to expose it - it should just be a minor revision. Would you like to submit a PR to expose the `XMLAttribute` `init`? I could get it merged and release with a `7.0.2` or similar.

**Reply by TkachenkoViacheslav** [reply]
It's cool! thanks)
Sorry, but I don't understand what is the "PR" abbreviation.
In my project I've made a workaround, and now I replace your version of the file with modified XMLAttribute.swift (in wich I'd added `public` keyword before init method) at the every cocoa pods install/update. So, I'll wait for the 7.0.2.
